src/view.c

```diff
@@ -333,7 +333,8 @@ goto_id(struct view *view, const char *expr, bool from_start, bool save_search)
 		report("Jumping to ID is not supported by the %s view", view->name);
 		return;
 	} else {
-		char *rev = argv_format_arg(view->env, expr);
+		char tmp[SIZEOF_STR];
+		char *rev = snprintf(tmp, SIZEOF_STR, "%s^{}", expr) ? argv_format_arg(view->env, tmp) : NULL;
```

Comment:
Looks good although the ? is weird (I don't think we'll ever get 0). Maybe ignore the return value explicitly using the comma operator:

```suggestion
		char *rev = (snprintf(tmp, SIZEOF_STR, "%s^{}", expr), argv_format_arg(view->env, tmp));
```

(The upcoming Git 2.35 has a "Weather balloon to break people with compilers that do not support C99.". Maybe we should try that too, then this will be more obvious)

Reply:
You're right, the test should have been at least `... > 0` but it doesn't make much sense here anyway. I was looking for other uses of the comma operator in Tig and found one strange one in `ui.c:303`. Looks like a typo for me, what do you think ?

As tig still works well with old Git versions, I think it is too early for us to think about dropping old compilers.